src/main/java/org/myrobotlab/io/StreamGobbler.java

```diff
@@ -4,28 +4,29 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.OutputStream;
 
 public class StreamGobbler extends Thread {
-  InputStream in;
-  String name;
-  boolean relay = true;
+  protected transient InputStream processOut;
+  protected transient OutputStream processIn;
+  protected String name;
 
-  public StreamGobbler(String name, InputStream is) {
+  public StreamGobbler(String name, InputStream processOut, OutputStream processIn) {
     super(name);
-    this.in = is;
+    this.processOut = processOut;
```

Comment:
What about stderr?  This will block if any of our 3rd parties write to stderr.  We don't want to bring back the mysterious dl4j locking up mrl again.